src/lib/queue_worker.js

```diff
@@ -113,7 +113,7 @@ QueueWorker.prototype._resetTask = function(taskRef, deferred) {
       } else {
         var errorMsg = 'reset task errored too many times, no longer retrying';
         logger.debug(self._getLogEntry(errorMsg), error);
-        deferred.reject(errorMsg);
+        deferred.reject(new Error(errorMsg));
```

Comment:
Event though the stack trace might not be helpful in all cases, it's nice just to be consistent. Also might help later on if more of the internal api is exposed and can be built upon.